lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,513 @@
+import { SwapDeal } from '../swaps/types';
+import { UnitConverter } from '../utils/UnitConverter';
+import { ConnextWallet } from './ConnextWallet';
+import errors, { errorCodes } from './errors';
+import {
+  Channel,
+  ConnextChannelCount,
+  ConnextClientConfig,
+  ConnextInfo,
+  ConnextVersion,
+  TokenPaymentRequest,
+  TokenPaymentResponse,
+} from './types';
+
+const MAX_AMOUNT = Number.MAX_SAFE_INTEGER;
+
+/**
+ * A class representing a client to interact with connext.
+ */
+class ConnextClient extends SwapClient {
+  public readonly type = SwapClientType.Connext;
+  public readonly finalLock = 100;
+  public address?: string;
+  /** A map of currency symbols to token addresses. */
+  public tokenAddresses = new Map<string, string>();
+  private network: string;
+  private ethProviderUrl: string;
+  private nodeUrl: string;
+  private unitConverter: UnitConverter;
+
+  private channel: IConnextClient | undefined;
+  private wallet: ConnextWallet | undefined;
+
+  /**
+   * Creates a connext client.
+   */
+  constructor({
+    config,
+    logger,
+    unitConverter,
+    currencyInstances,
+  }: {
+    unitConverter: UnitConverter;
+    config: ConnextClientConfig;
+    currencyInstances: CurrencyInstance[],
+    logger: Logger;
+  }) {
+    super(logger, config.disable);
+
+    this.network = config.network || '';
+    this.ethProviderUrl = config.ethProviderUrl || '';
+    this.nodeUrl = config.nodeUrl || '';
+    this.unitConverter = unitConverter;
+    this.setTokenAddresses(currencyInstances);
+  }
+
+  public get minutesPerBlock() {
+    return 0.25; // 15 seconds per block target
+  }
+
+  public get label() {
+    return 'Connext';
+  }
+
+  /**
+   * Checks for connectivity and gets our Connext account address
+   */
+  public initSpecific = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    if (this.wallet) {
+      await this.initConnext();
+      this.setStatus(ClientStatus.Initialized);
+      this.emit('initialized');
+      await this.verifyConnectionWithTimeout();
+    }
+  }
+  /**
+   * Initiates wallet for the Connext client
+   */
+  public initWallet = async (seedMnemonic: string[]) => {
+    this.wallet = new ConnextWallet(seedMnemonic.join(' '));
+  }
+
+  public initConnext = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    const wallet = this.wallet;
+    const channel = await connext.connect(this.network, {
+      ethProviderUrl: this.ethProviderUrl,
+      nodeUrl: this.nodeUrl,
+      xpub: wallet.xpub,
+      keyGen: (index: string) => wallet.keyGen(index),
+      store: new ConnextStore(new FileStorage()),
+    });
+    this.channel = channel;
+    return channel;
+  }
+  /**
+   * Associate connext with currencies that have a token address
+   */
+  private setTokenAddresses = (currencyInstances: CurrencyInstance[]) => {
+    currencyInstances.forEach((currency) => {
+      if (currency.tokenAddress) {
```

Comment:
If we decide to go with 1 swap client per currency then this will also need a check for `SwapClient.Connext`.

Reply:
> if we decide to go with 1 swap client per currency

definitely one swap client per currency, see below

Reply:
On a side note (and not needed to be done in this PR) it would be good to have the same logic in RaidenClient, or actually better yet within SwapClientManager. I don't think this needs to be done as part of this PR though, I can take a look at implementing it separately.

Reply:
As per the other comment, WETH and DAI will use Raiden and ETH and USDT will use Connext